File: hw6_problem10.py
import math
import random

def get_D_n(n):
    # TODO: implement this function to sample D_n for a given n
    # You may need to generate X_1, ..., X_n ~ i.i.d. U(0,1) first and then compute D_n.
    return 0

if __name__ == '__main__':
    random.seed(1898)

    print(f"n\tDn\tln(n)Dn\tsqrt(n)Dn\tnDn")

    for n in [10**z for z in range(1,8)]:
        D_n = get_D_n(n)
        print(f"10^{math.log10(n):.0f}\t{D_n:.4f}\t{math.log(n)*D_n:.4f}\t{math.sqrt(n)*D_n:.4f}\t\t{n*D_n:.4f}")

File: hw6problem10.cpp
#include <bits/stdc++.h>

double sample_uniform(double low, double high){
    /* Helper function: sample a random variable from U(low,high) distribution. */
    return ((double) rand() / RAND_MAX) * (high-low) + low;
}

double get_D_n(int n){
    // TODO: implement this function to sample D_n for a given n
    // You may need to generate X_1, ..., X_n ~ i.i.d. U(0,1) first and then compute D_n.
    return 0;
}

int main(){
    srand(1898);

    printf("n\tDn\tln(n)Dn\tsqrt(n)Dn\tnDn\n");
    for (int i=1,n=10; i<8; ++i,n*=10){
        double D_n = get_D_n(n);
        printf("10^%d\t%.4f\t%.4f\t%.4f\t\t%.4f\n", 
                i, D_n, log(n)*D_n, sqrt(n)*D_n, n*D_n);
    }
    return 0;
}